### ISSUE TITLE
build failing on M1 Mac

### ISSUE THREAD

**Issue by adokter** [adokter/vol2birdR#27]
After refactoring of the build system in recent commits the proj library can no longer be found on this M1 Max machine. See attached install_log.txt for more info

```
12.6.1 Monterey
Apple M1 Max 32 Gb
```

```
R.version
               _
platform       aarch64-apple-darwin20
arch           aarch64
os             darwin20
system         aarch64, darwin20
status
major          4
minor          2.1
year           2022
month          06
day            23
svn rev        82513
language       R
version.string R version 4.2.1 (2022-06-23)
nickname       Funny-Looking Kid
```

[install_log.txt](https://github.com/adokter/vol2birdR/files/10134299/install_log.txt)

**Comment by adokter** [comment]
Installing `pkg-config` solved this issue. @andershenja do we want to add pkg-config as a required dependency, or does this need more work?

**Comment by PietrH** [comment]
I had a similar problem, also solved by installing pkg-config.

My session info: 

```md
─ Session info ─────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.2.1 (2022-06-23)
 os       Ubuntu 22.10
 system   x86_64, linux-gnu
 ui       RStudio
 language en_US
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       Europe/Brussels
 date     2023-05-15
 rstudio  2023.03.0+386 Cherry Blossom (desktop)
 pandoc   NA

─ Packages ─────────────────────────────────────────────────────────────────
 package     * version date (UTC) lib source
 cachem        1.0.8   2023-05-01 [1] CRAN (R 4.2.1)
 callr         3.7.3   2022-11-02 [1] CRAN (R 4.2.1)
 cli           3.6.1   2023-03-23 [1] CRAN (R 4.2.1)
 crayon        1.5.2   2022-09-29 [1] CRAN (R 4.2.1)
 curl          5.0.0   2023-01-12 [1] CRAN (R 4.2.1)
 desc          1.4.2   2022-09-08 [1] CRAN (R 4.2.1)
 devtools      2.4.5   2022-10-11 [1] CRAN (R 4.2.1)
 digest        0.6.31  2022-12-11 [1] CRAN (R 4.2.1)
 ellipsis      0.3.2   2021-04-29 [1] CRAN (R 4.2.1)
 fastmap       1.1.1   2023-02-24 [1] CRAN (R 4.2.1)
 fs            1.6.2   2023-04-25 [1] CRAN (R 4.2.1)
 glue          1.6.2   2022-02-24 [1] CRAN (R 4.2.1)
 htmltools     0.5.5   2023-03-23 [1] CRAN (R 4.2.1)
 htmlwidgets   1.6.2   2023-03-17 [1] CRAN (R 4.2.1)
 httpuv        1.6.11  2023-05-11 [1] CRAN (R 4.2.1)
 later         1.3.1   2023-05-02 [1] CRAN (R 4.2.1)
 lifecycle     1.0.3   2022-10-07 [1] CRAN (R 4.2.1)
 magrittr      2.0.3   2022-03-30 [1] CRAN (R 4.2.1)
 memoise       2.0.1   2021-11-26 [1] CRAN (R 4.2.1)
 mime          0.12    2021-09-28 [1] CRAN (R 4.2.1)
 miniUI        0.1.1.1 2018-05-18 [1] CRAN (R 4.2.1)
 pkgbuild      1.4.0   2022-11-27 [1] CRAN (R 4.2.1)
 pkgload       1.3.2   2022-11-16 [1] CRAN (R 4.2.1)
 prettyunits   1.1.1   2020-01-24 [1] CRAN (R 4.2.1)
 processx      3.8.1   2023-04-18 [1] CRAN (R 4.2.1)
 profvis       0.3.8   2023-05-02 [1] CRAN (R 4.2.1)
 promises      1.2.0.1 2021-02-11 [1] CRAN (R 4.2.1)
 ps            1.7.5   2023-04-18 [1] CRAN (R 4.2.1)
 purrr         1.0.1   2023-01-10 [1] CRAN (R 4.2.1)
 R6            2.5.1   2021-08-19 [1] CRAN (R 4.2.1)
 Rcpp          1.0.10  2023-01-22 [1] CRAN (R 4.2.1)
 remotes       2.4.2   2021-11-30 [1] CRAN (R 4.2.1)
 rlang         1.1.1   2023-04-28 [1] CRAN (R 4.2.1)
 rprojroot     2.0.3   2022-04-02 [1] CRAN (R 4.2.1)
 rstudioapi    0.14    2022-08-22 [1] CRAN (R 4.2.1)
 sessioninfo   1.2.2   2021-12-06 [1] CRAN (R 4.2.1)
 shiny         1.7.4   2022-12-15 [1] CRAN (R 4.2.1)
 stringi       1.7.12  2023-01-11 [1] CRAN (R 4.2.1)
 stringr       1.5.0   2022-12-02 [1] CRAN (R 4.2.1)
 urlchecker    1.0.1   2021-11-30 [1] CRAN (R 4.2.1)
 usethis       2.1.6   2022-05-25 [1] CRAN (R 4.2.1)
 vctrs         0.6.2   2023-04-19 [1] CRAN (R 4.2.1)
 withr         2.5.0   2022-03-03 [1] CRAN (R 4.2.1)
 xtable        1.8-4   2019-04-21 [1] CRAN (R 4.2.1)
```


Installation log: [install_log.txt](https://github.com/adokter/vol2birdR/files/11476480/install_log.txt)

The relevant part being:

```sh
Error: package or namespace load failed for ‘vol2birdR’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/home/pieter_huybrechts/R/x86_64-pc-linux-gnu-library/4.2/00LOCK-vol2birdR/00new/vol2birdR/libs/vol2birdR.so':
  /home/pieter_huybrechts/R/x86_64-pc-linux-gnu-library/4.2/00LOCK-vol2birdR/00new/vol2birdR/libs/vol2birdR.so: undefined symbol: pj_release
Error: loading failed
Execution halted
ERROR: loading failed
```

Running on an intel machine.

**Comment by adokter** [comment]
Thanks @PietrH, do you happen to have `config.log` file as well for the case in which your build fails? When pkg-config is not found it should revert to using homebrew instead.

Homebrew requires setting some things to your path, see bottom of https://github.com/adokter/vol2birdinstall, did you complete those steps? We should document this more clearly in vol2birdR installation instructions as well

**Comment by PietrH** [comment]
I'm afraid I do not, at least I didn't store my system state manually anywhere. I actually went trough quite a few steps to get it all to work, including installing gfortran 11.

I don't currently have brew installed, I'm running Ubuntu 22.10.

**Comment by adokter** [comment]
ah gotcha - on ubuntu pkg-config should be a system requirement on linux systems, we have to clarify that better

**Comment by adokter** [comment]
@PietrH any idea why `pkg-config` was not present on your Ubuntu machine? I thought it pre-installed: https://manpages.ubuntu.com/manpages/kinetic/man1/pkg-config.1.html

**Comment by PietrH** [comment]
I have no idea, apart from pkg-config itself some perl dependencies were installed alongside.

```go
$ sudo apt-get -y install pkg-config

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  libdpkg-perl libfile-fcntllock-perl
Suggested packages:
  debian-keyring bzr dpkg-dev
The following NEW packages will be installed:
  libdpkg-perl libfile-fcntllock-perl pkg-config
0 upgraded, 3 newly installed, 0 to remove and 13 not upgraded.
Need to get 319 kB of archives.
After this operation, 2.722 kB of additional disk space will be used.
Get:1 http://be.archive.ubuntu.com/ubuntu kinetic/main amd64 libdpkg-perl all 1.21.9ubuntu1 [237 kB]
Get:2 http://be.archive.ubuntu.com/ubuntu kinetic/main amd64 libfile-fcntllock-perl amd64 0.22-4 [33,7 kB]
Get:3 http://be.archive.ubuntu.com/ubuntu kinetic/main amd64 pkg-config amd64 0.29.2-1ubuntu3 [48,2 kB]
Fetched 319 kB in 0s (1.195 kB/s)    
Selecting previously unselected package libdpkg-perl.
(Reading database ... 269745 files and directories currently installed.)
Preparing to unpack .../libdpkg-perl_1.21.9ubuntu1_all.deb ...
Unpacking libdpkg-perl (1.21.9ubuntu1) ...
Selecting previously unselected package libfile-fcntllock-perl.
Preparing to unpack .../libfile-fcntllock-perl_0.22-4_amd64.deb ...
Unpacking libfile-fcntllock-perl (0.22-4) ...
Selecting previously unselected package pkg-config.
Preparing to unpack .../pkg-config_0.29.2-1ubuntu3_amd64.deb ...
Unpacking pkg-config (0.29.2-1ubuntu3) ...
Setting up libfile-fcntllock-perl (0.22-4) ...
Setting up libdpkg-perl (1.21.9ubuntu1) ...
Setting up pkg-config (0.29.2-1ubuntu3) ...
Processing triggers for man-db (2.10.2-2) ...
```

**Comment by adokter** [comment]
fixed by #46
